code/game/objects/items/weapons/butterfly.dm

```diff
@@ -45,7 +45,7 @@
 		fold()
 		to_chat(L, "You flip \the [src] closed.")
 		if(bug)
-			if(bug == (/mob/living/simple_animal/hostile/viscerator/butterfly || /mob/living/simple_animal/hostile/viscerator/butterfly/magic))
+			if((bug == /mob/living/simple_animal/hostile/viscerator/butterfly) || (bug == /mob/living/simple_animal/hostile/viscerator/butterfly/magic))
```

Comment:
Just check if(istype(bug, /mob/living/simple_animal/hostile/viscerator/butterfly))

Reply:
This doesn't work.